### PR TITLE
Add docs for CSR signing.

### PR DESCRIPTION
**Pull request by ncabatoff**

### CODE REVIEW

**Comment by ncabatoff** [comment]
Making it a draft so we don't accidentally merge it before 1.5.

File: website/pages/api-docs/secret/kmip/index.mdx [hunk 1]
```diff
@@ -428,6 +428,66 @@ $ curl \
...
+  and CA chain in. One of `pem`, `pem_bundle`, or `der`.
+- `csr` (`string`) - CSR in PEM format.
+
+### Sample Request
+
+```
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request POST \
+    --data '{"csr": "-----BEGIN CERTIFICATE REQUEST-----
```
Comment (calvn) [inline website/pages/api-docs/secret/kmip/index.mdx:455]: Can the JSON value be provided like so, or do the newlines need to be `\n`?

File: website/pages/docs/secrets/kmip/index.mdx [hunk 2]
```diff
@@ -133,6 +133,68 @@ which will be used when evaluating permissions during a KMIP request.
       serial_number    317328055225536560033788492808123425026102524390
     ```
 
+### Client Certificate Signing
+
+As an alternative to the above section on generating client certificates,
+the KMIP secrets engine supports signing of Certificate Signing Requests
+(CSRs). Normally the above generation process is simpler, but some KMIP
+clients prefer (or only support) to retain the private key associated
+with their client certificate.0
```
Comment (calvn) [inline website/pages/docs/secrets/kmip/index.mdx:142]:
```suggestion
with their client certificate.
```

File: website/pages/docs/secrets/kmip/index.mdx [hunk 3]
```diff
@@ -133,6 +133,68 @@ which will be used when evaluating permissions during a KMIP request.
       serial_number    317328055225536560033788492808123425026102524390
     ```
 
+### Client Certificate Signing
+
+As an alternative to the above section on generating client certificates,
+the KMIP secrets engine supports signing of Certificate Signing Requests
+(CSRs). Normally the above generation process is simpler, but some KMIP
+clients prefer (or only support) to retain the private key associated
```
Comment (calvn) [inline website/pages/docs/secrets/kmip/index.mdx:141]:
```suggestion
clients prefer (or only support) retaining the private key associated
```

Comment (calvn) [reply website/pages/docs/secrets/kmip/index.mdx:141]: Alternative: `but some KMIP clients prefer to retain (or only support) ...`

File: website/pages/api-docs/secret/kmip/index.mdx (hunk 1 above)
Comment (ncabatoff) [reply website/pages/api-docs/secret/kmip/index.mdx:455]: Good catch, you're right they must be `\n`.

**Pull request merged**